### PR TITLE
daemon: switch the /snaps endpoint to switch to /find if it includes the (undocumented) "sources=store"

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

**Comment by chipaca** [comment]
retest this please

**Comment by pedronis** [comment]
btw do we know/have we talked with the gnome software center people about these changes to /snaps vs find? aren't they consumed these bits as well? or are they talking to the store directly?

**Comment by chipaca** [comment]
retest this please

**Comment by chipaca** [comment]
@pedronis that's a good point. And we *are* going to have a problem.

**Comment by chipaca** [comment]
retest this please

**Comment by chipaca** [comment]
retest this please

File: daemon/api.go [hunk 1]
```diff
@@ -392,17 +375,18 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
...
 	}
 
 	results := make([]*json.RawMessage, len(found))
 	for i, x := range found {
-		resource := ""
 		url, err := route.URL("name", x.Name())
-		if err == nil {
-			resource = url.String()
+		if err != nil {
+			logger.Noticef("unable to build URL for snap %q (r%d): %v", x.Name(), x.Revision, err)
```
Comment (zyga) [inline daemon/api.go]: `cannot build URL for snap %q (revision %d): %v`

**Comment by chipaca** [comment]
retest this please

File: daemon/api.go [hunk 2]
```diff
@@ -413,126 +397,68 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
...
+		logger.Noticef("jumping to \"find\" to better support legacy request %q", r.URL)
+		return searchStore(c, r, user)
+	}
 
-		for _, snap := range found {
-			remoteSnapMap[snap.Name()] = snap
-		}
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("router can't find route for snaps")
```
Comment (zyga) [inline daemon/api.go]: `cannot find router for snaps`

File: daemon/api.go [hunk 3]
```diff
@@ -413,126 +397,68 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
...
+		if err != nil {
+			logger.Noticef("unable to build URL for snap %q (r%d): %v", name, rev, err)
+			continue
 		}
 
-		data, err := json.Marshal(webify(m, resource))
+		data, err := json.Marshal(webify(mapLocal(x.info, isActive), url.String()))
 		if err != nil {
-			return
+			return InternalError("unable to serialize snap %q (r%d): %v", name, rev, err)
```
Comment (zyga) [inline daemon/api.go]: `cannot serialize snap %q ...`

**Comment by zyga** [comment]
The code looks good, +1

File: client/packages.go [hunk 4]
```diff
@@ -66,34 +58,44 @@ const (
...
-	if filter.Query != "" {
-		q.Set("q", filter.Query)
+	if len(names) == 0 {
+		return snaps, nil
 	}
 
-	if len(filter.Sources) > 0 {
-		q.Set("sources", strings.Join(filter.Sources, ","))
+	wanted := make(map[string]bool, len(names))
+	for i := range names {
```
Comment (zyga) [inline client/packages.go]: Perhaps (bike shed should be blue)
```
for _, name := range names {
    wanted[name] = true
}
```

**Comment by zyga** [comment]
Looks goodish +1

**Comment by pedronis** [comment]
LGTM

**Pull request merged**